### PR TITLE
ensure only one key is deleted if multiple with same name

### PR DESCRIPTION
**Pull request by TylerNoblett**
Ensure that if there are multiple keys with the same name, only one is deleted

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: backend/helpers/functional.go [hunk 1]
```diff
@@ -16,7 +16,7 @@ func Map[T any, U any](slice []T, mapFn func(T) U) []U {
 	return result
 }
 
-// Find is a generic function that searches through a list searching for all items that match
+// Filter is a generic function that searches through a list searching for all items that match
```
Comment (JoelAtDeluxe) [inline backend/helpers/functional.go:19]: Oops!

**Review by jrozner — CHANGES_REQUESTED** [review]
It looks like this would remove the first key rather than the one selected. I think it would make sense to either add a unique constraint on the name (so this problem doesn't exist) or swap out the identifier in the DELETE request to be a unique identifier (like we do with slugs) so that we can target them directly.

**Comment by TylerNoblett** [comment]
> It looks like this would remove the first key rather than the one selected. I think it would make sense to either add a unique constraint on the name (so this problem doesn't exist) or swap out the identifier in the DELETE request to be a unique identifier (like we do with slugs) so that we can target them directly.

oh that's a good point - yeah I feel like the first one makes the most sense. I'll take a look

**Comment by TylerNoblett** [comment]
@jrozner we're now deleting based on the unique credential ID instead of the name

**Review by jkennedyvz — APPROVED** [review]

**Comment by TylerNoblett** [comment]
@jrozner @jkennedyvz I just added a check to ensure that the username is unique

**Review by jrozner — APPROVED** [review]

**Pull request merged**